Update to go1.11.1

cblecker opened the pull request:
**What this PR does / why we need it**:
Updates kubernetes to build under go1.11.1

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #67872

**Special notes for your reviewer**:
Will need CI images/environments to be bumped too, as there are gofmt changes.

**Release note**:
```release-note
Update to use go1.11.1
```

/assign @ixdy @BenTheElder 
/cc @dims @kubernetes/sig-release-misc @kubernetes/sig-scalability-misc

cblecker commented:
oh, also cc: @kubernetes/sig-testing regarding CI bumps

timothysc commented:
/assign @shyamjvs 

@kubernetes/sig-scalability-feature-requests typically owns the 👍 👎  on go versions.

BenTheElder reviewed (approved):
/lgtm
:tada:

ixdy commented on build/root/WORKSPACE:
```diff
@@ -3,8 +3,9 @@ load("//build:workspace.bzl", "CRI_TOOLS_VERSION")
 http_archive(
     name = "io_bazel_rules_go",
-    sha256 = "97cf62bdef33519412167fd1e4b0810a318a7c234f5f8dc4f53e2da86241c492",
-    urls = mirror("https://github.com/bazelbuild/rules_go/releases/download/0.15.3/rules_go-0.15.3.tar.gz"),
```
rules_go 0.15.4 is out now with support for go1.11.1.

cblecker replied:
Updated!

BenTheElder commented:
working on the CI image bumps

cblecker committed:
Bump rules_go to 0.15.4 to support go1.11.1

cblecker committed:
Bump golang version to 1.11.1

cblecker committed:
Fix TestGetNameFromCallsite as the test runner line changed.

https://github.com/golang/go/blob/release-branch.go1.10/src/testing/testing.go#L777
https://github.com/golang/go/blob/release-branch.go1.11/src/testing/testing.go#L827

ixdy commented on build/build-image/cross/VERSION:
```diff
@@ -1 +1 @@
-v1.10.4-1
+v1.11.1-1
```
I've built and pushed k8s.gcr.io/kube-cross:v1.11.1-1. Waiting for it to sync to mirrors.

ixdy commented on hack/lib/golang.sh:
```diff
@@ -360,7 +360,7 @@ EOF
   IFS=" " read -ra go_version <<< "$(go version)"
   local minimum_go_version
-  minimum_go_version=go1.10.2
+  minimum_go_version=go1.11.1
```
hm, lots of verify checks fail unless verify is bumped first, but bumping verify first means that any other PR will fail (due to gofmt failures).

ixdy commented:
/retest

BenTheElder commented:
note: verify is going to fail in some capacity once we merge the revert in https://github.com/kubernetes/test-infra/pull/9691

some of the failures look legitimate :(

ixdy commented:
The bazel-test failure is a bit of a mess. https://github.com/kubernetes/repo-infra/issues/86

cblecker commented:
/retest

jennybuckley commented:
/assign @lavalamp

cblecker committed:
Bump repo-infra to e8f2f7c

cblecker commented:
/test pull-kubernetes-bazel-test

cblecker commented:
The current bazel test failure may be a bazel specific rules_go issue.
- bazel log from CI (`bazel test //staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/envelope:go_default_test
`):
https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/pr-logs/pull/69386/pull-kubernetes-bazel-test/62546/#stagingsrck8sioapiserverpkgstoragevalueencryptenvelopego_default_test
- my local output from go-native 1.11 testing:
```
$ KUBE_RUN_COPY_OUTPUT=n build/run.sh go test -race -v ./staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/envelope
+++ [1004 15:47:59] Verifying Prerequisites....
+++ [1004 15:47:59] Using Docker for MacOS
+++ [1004 15:47:59] Building Docker image kube-build:build-4595eb51c1-5-v1.11.1-1
+++ [1004 15:49:32] Keeping image kube-build:build-4595eb51c1-5-v1.11.1-1
+++ [1004 15:49:32] Creating data container kube-build-data-4595eb51c1-5-v1.11.1-1
+++ [1004 15:49:39] Syncing sources to container
+++ [1004 15:49:40] Stopping any currently running rsyncd container
+++ [1004 15:49:40] Starting rsyncd container
+++ [1004 15:49:41] Running rsync
+++ [1004 15:49:55] Stopping any currently running rsyncd container
+++ [1004 15:49:56] Output from this container will NOT be rsynced out upon completion. Set KUBE_RUN_COPY_OUTPUT=y to enable.
+++ [1004 15:49:56] Running build command...
=== RUN   TestEnvelopeCaching
--- PASS: TestEnvelopeCaching (0.00s)
=== RUN   TestEnvelopeCacheLimit
--- PASS: TestEnvelopeCacheLimit (0.00s)
=== RUN   TestBackwardsCompatibility
--- PASS: TestBackwardsCompatibility (0.00s)
=== RUN   TestKMSPluginLateStart
E1004 15:50:05.553044    1493 grpc_service.go:71] failed to create connection to unix socket: @kms-socket.sock, error: dial unix @kms-socket.sock: connect: connection refused
E1004 15:50:06.554492    1493 grpc_service.go:71] failed to create connection to unix socket: @kms-socket.sock, error: dial unix @kms-socket.sock: connect: connection refused
--- PASS: TestKMSPluginLateStart (2.68s)
=== RUN   TestIntermittentConnectionLoss
E1004 15:50:08.235518    1493 grpc_service.go:71] failed to create connection to unix socket: @kms-socket.sock, error: dial unix @kms-socket.sock: connect: connection refused
--- PASS: TestIntermittentConnectionLoss (1.01s)
    grpc_service_unix_test.go:93: Connected to KMSPlugin
    grpc_service_unix_test.go:97: KMS Plugin is stopped
    grpc_service_unix_test.go:104: Sending encrypt request
    grpc_service_unix_test.go:120: Restarted KMS Plugin
=== RUN   TestUnsupportedVersion
--- PASS: TestUnsupportedVersion (0.01s)
=== RUN   TestConcurrentAccess
--- PASS: TestConcurrentAccess (0.00s)
=== RUN   TestGRPCService
--- PASS: TestGRPCService (0.01s)
=== RUN   TestGRPCServiceConcurrentAccess
--- PASS: TestGRPCServiceConcurrentAccess (1.35s)
=== RUN   TestInvalidConfiguration
=== RUN   TestInvalidConfiguration/emptyConfiguration
=== RUN   TestInvalidConfiguration/invalidScheme
--- PASS: TestInvalidConfiguration (0.00s)
    --- PASS: TestInvalidConfiguration/emptyConfiguration (0.00s)
    --- PASS: TestInvalidConfiguration/invalidScheme (0.00s)
PASS
ok      k8s.io/kubernetes/staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/envelope       6.124s
```

cc: @ixdy

ixdy commented:
hm, the second bazel-test invocation passed. Could it maybe be the test? It was only added about a week ago in #68585.

(OTOH, it isn't showing any failures in http://k8s-testgrid.appspot.com/sig-release-master-blocking#bazel-test&sort-by-flakiness=&width=5.)

cc @immutableT

BenTheElder commented:
That test uses `sleep` which immediately sets off my flake-radar :^)

ixdy commented:
Ah, good sleuthing. Also, the CI bazel-test job may not be the best metric, since I think a lot of tests end up being cached there.

Looking at [presubmit bazel-test](http://k8s-testgrid.appspot.com/presubmits-kubernetes-blocking#pull-kubernetes-bazel-test&sort-by-flakiness=), this is the third most flakiest test, with similar failures as we saw here. So I don't think this is a go1.11.1 issue.

immutableT commented:
It may be that envelope test is a bit too ambitions - trying to simulate scenarios of recovering from  intermittent network failures, and it is timing dependent. I will take a look at it in the morning.

immutableT commented:
69470 increases time-out in staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/envelope; this should result in the reduction of intermittent failures. I will continue monitoring this test.

lavalamp commented:
Happy to approve once the tests look a bit greener :)

ixdy commented:
I'm not sure exactly how to proceed here.

Most of the test failures are because the minimum go version check requires go1.11.1.

We could punt that to a separate PR, though the verify job will probably still fail, due to `gofmt` changes.

If we bump the images to go1.11.1, then this PR will pass, but the verify job will fail all other PRs in master for a bit.

cblecker commented:
Because the fmt changes and the TestGetNameFromCallsite fix both require go1.11, I think they should all stay in the same PR.

I think we just need to pick a time, bump the test-infra images, and then retest this for merge.

ixdy commented:
@lavalamp would this^ proposal work for you?

cblecker committed:
Update gofmt for go1.11

lavalamp commented:
Yes, that works for me as long as we warn people (message the dev mailing
list?) and we're pretty confident that we won't find some other test
failure that's currently masked.

On Fri, Oct 5, 2018 at 12:53 PM Jeff Grafton <notifications@github.com>
wrote:

> @lavalamp <https://github.com/lavalamp> would this^ proposal work for you?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/69386#issuecomment-427480078>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAnglp2PrrnhDKWx3Xs8mcN_ouPi3Dj7ks5uh7jOgaJpZM4XGjKF>
> .
>

cblecker commented:
Rebased again to resolve conflicts with the gofmt stuff

BenTheElder commented:
I'm ready to switch the images back to go1.11 from the infra side any time, please let me know. IMO we should get this done. I think we should mail the ML _after_ we switch to go1.11 and managed to merge this, otherwise we'll just be spamming them when we have to revert.

BenTheElder commented:
We also should not find any other failures, a lot of the tests use bazel which will switch to go1.11 without any image changes, out of the jobs that are failing:
- verify is failing due to not having 1.11. we switched breifly the other day to 1.11 on accident, and it was fine
- typecheck is failing due to the go1.11 version check
- local-e2e-containerized is generically broken currently and does not block PRs
- node-e2e is failing the go1.11 version check
- integration is failing the version check, and is eternally flaky so we can't get a good signal anyhow :man_shrugging:

cblecker commented:
SGTM.

@lavalamp: Can we get an approve then, based on the above?

BenTheElder commented:
Image bump is ready in https://github.com/kubernetes/test-infra/pull/9716

lavalamp commented:
/approve

/hold

unhold when ready!

lavalamp commented:
also

/lgtm

BenTheElder commented:
/retest

lavalamp commented:
/hold cancel

I guess that wasn't really necessary.

lavalamp commented:
I sent a heads-up to the kubernetes-dev mailing list, since this will be disruptive.

BenTheElder commented:
I think integration may have flaked, but that's kinda to be expected with integration lately :/

BenTheElder commented:
/retest

lavalamp commented:
@cblecker @BenTheElder looks like an actual go vet failure.

Fix asap, merge while broken & quick follow-up, or abandon ship?

cblecker committed:
fix vet error in pkg/controller/route/route_controller_test.go

pkg/controller/route/route_controller_test.go:101: *k8s.io/kubernetes/vendor/k8s.io/cloud-provider.Route composite literal uses unkeyed fields
pkg/controller/route/route_controller_test.go:102: *k8s.io/kubernetes/vendor/k8s.io/cloud-provider.Route composite literal uses unkeyed fields
pkg/controller/route/route_controller_test.go:105: *k8s.io/kubernetes/vendor/k8s.io/cloud-provider.Route composite literal uses unkeyed fields
pkg/controller/route/route_controller_test.go:106: *k8s.io/kubernetes/vendor/k8s.io/cloud-provider.Route composite literal uses unkeyed fields
pkg/controller/route/route_controller_test.go:118: *k8s.io/kubernetes/vendor/k8s.io/cloud-provider.Route composite literal uses unkeyed fields
pkg/controller/route/route_controller_test.go:121: *k8s.io/kubernetes/vendor/k8s.io/cloud-provider.Route composite literal uses unkeyed fields
pkg/controller/route/route_controller_test.go:122: *k8s.io/kubernetes/vendor/k8s.io/cloud-provider.Route composite literal uses unkeyed fields
pkg/controller/route/route_controller_test.go:135: *k8s.io/kubernetes/vendor/k8s.io/cloud-provider.Route composite literal uses unkeyed fields
pkg/controller/route/route_controller_test.go:136: *k8s.io/kubernetes/vendor/k8s.io/cloud-provider.Route composite literal uses unkeyed fields
pkg/controller/route/route_controller_test.go:148: *k8s.io/kubernetes/vendor/k8s.io/cloud-provider.Route composite literal uses unkeyed fields
pkg/controller/route/route_controller_test.go:149: *k8s.io/kubernetes/vendor/k8s.io/cloud-provider.Route composite literal uses unkeyed fields
pkg/controller/route/route_controller_test.go:150: *k8s.io/kubernetes/vendor/k8s.io/cloud-provider.Route composite literal uses unkeyed fields
pkg/controller/route/route_controller_test.go:151: *k8s.io/kubernetes/vendor/k8s.io/cloud-provider.Route composite literal uses unkeyed fields
pkg/controller/route/route_controller_test.go:154: *k8s.io/kubernetes/vendor/k8s.io/cloud-provider.Route composite literal uses unkeyed fields
pkg/controller/route/route_controller_test.go:155: *k8s.io/kubernetes/vendor/k8s.io/cloud-provider.Route composite literal uses unkeyed fields
pkg/controller/route/route_controller_test.go:167: *k8s.io/kubernetes/vendor/k8s.io/cloud-provider.Route composite literal uses unkeyed fields
pkg/controller/route/route_controller_test.go:168: *k8s.io/kubernetes/vendor/k8s.io/cloud-provider.Route composite literal uses unkeyed fields
pkg/controller/route/route_controller_test.go:171: *k8s.io/kubernetes/vendor/k8s.io/cloud-provider.Route composite literal uses unkeyed fields
pkg/controller/route/route_controller_test.go:172: *k8s.io/kubernetes/vendor/k8s.io/cloud-provider.Route composite literal uses unkeyed fields
pkg/controller/route/route_controller_test.go:185: *k8s.io/kubernetes/vendor/k8s.io/cloud-provider.Route composite literal uses unkeyed fields
pkg/controller/route/route_controller_test.go:197: *k8s.io/kubernetes/vendor/k8s.io/cloud-provider.Route composite literal uses unkeyed fields
pkg/controller/route/route_controller_test.go:198: *k8s.io/kubernetes/vendor/k8s.io/cloud-provider.Route composite literal uses unkeyed fields
pkg/controller/route/route_controller_test.go:199: *k8s.io/kubernetes/vendor/k8s.io/cloud-provider.Route composite literal uses unkeyed fields
pkg/controller/route/route_controller_test.go:202: *k8s.io/kubernetes/vendor/k8s.io/cloud-provider.Route composite literal uses unkeyed fields
pkg/controller/route/route_controller_test.go:203: *k8s.io/kubernetes/vendor/k8s.io/cloud-provider.Route composite literal uses unkeyed fields
pkg/controller/route/route_controller_test.go:215: *k8s.io/kubernetes/vendor/k8s.io/cloud-provider.Route composite literal uses unkeyed fields
pkg/controller/route/route_controller_test.go:216: *k8s.io/kubernetes/vendor/k8s.io/cloud-provider.Route composite literal uses unkeyed fields
pkg/controller/route/route_controller_test.go:217: *k8s.io/kubernetes/vendor/k8s.io/cloud-provider.Route composite literal uses unkeyed fields
pkg/controller/route/route_controller_test.go:220: *k8s.io/kubernetes/vendor/k8s.io/cloud-provider.Route composite literal uses unkeyed fields
pkg/controller/route/route_controller_test.go:221: *k8s.io/kubernetes/vendor/k8s.io/cloud-provider.Route composite literal uses unkeyed fields
pkg/controller/route/route_controller_test.go:222: *k8s.io/kubernetes/vendor/k8s.io/cloud-provider.Route composite literal uses unkeyed fields

cblecker committed:
fix vet error in pkg/kubelet/kubelet_test.go

pkg/kubelet/kubelet_test.go:1573: *k8s.io/kubernetes/pkg/kubelet/container/testing.FakePod composite literal uses unkeyed fields

BenTheElder commented:
fix asap, if we abandon ship and go back to go1.10 more of these will be introduced, these are recent changes in master that don't work for 1.11

BenTheElder commented:
they also look like very trivial fixes hopefully. they're all in _our_ code, and they're small stuff like printf mistakes

cblecker committed:
fix vet error in plugin/pkg/admission/storage/persistentvolume/label/admission.go

plugin/pkg/admission/storage/persistentvolume/label/admission.go:173: Verbose.Info call has possible formatting directive %v

lavalamp commented:
OK, we'll wait for a fix in this PR then :)

BenTheElder commented on test/soak/serve_hostnames/serve_hostnames.go:
```diff
@@ -116,12 +116,12 @@ func main() {
 	got, err := client.CoreV1().Namespaces().Create(&v1.Namespace{ObjectMeta: metav1.ObjectMeta{GenerateName: "serve-hostnames-"}})
 	if err != nil {
-		glog.Fatalf("Failed to create namespace: %v", err)
+		glog.Fatalf("Failed to create namespace: %e", err)
```
this doesn't seem right?

cblecker committed:
fix vet error in test/soak/cauldron/cauldron.go

test/soak/cauldron/cauldron.go:106: Warningf format %e has arg ns of wrong type string

cblecker committed:
fix vet error in test/soak/serve_hostnames/serve_hostnames.go

test/soak/serve_hostnames/serve_hostnames.go:124: Warningf format %e has arg ns of wrong type string

BenTheElder reviewed (approved):
/lgtm
